Ahmedq02/task-app
# Database seeding

# Database seeding migration



## Description

Add initial data to the database, for tasks and passkeys

- **Related Issue:** Closes #8 
- **Summary of Changes:**
  - Create CSV with initial data
  - Create custom migration to load the CSV data into the tasks database
  - Create command to create a passkey user from the environment variables

## How to Test

1. Check out this branch: `git checkout <branch-name>`
2. Delete all tasks from the database
3. Add new variables to .env file
4. If the passkey is not present, run  `poetry run python manage.py createpasskey`
5. Apply migrations: `migrate`
6. Run the development server: `make run`
7. Test the following:
   - [ ] Tasks from CSV are present 

## Deployment Notes



- [x] This PR requires database migrations
- [x] New environment variables need to be added: PASSKEY - the password for accessing the website
